File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
    data := make([]int, 0, 10)
	postorder(root, &data)
	return data
}

func postorder(root *TreeNode, data *[]int) {
	if root == nil {
		return
	}

	postorder(root.Left, data)
	postorder(root.Right, data)

	*data = append(*data, root.Val)
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        data = []
        
        self.prepare(root, data)

        return data

    def prepare(self, root: Optional[TreeNode], data: List[int]):
        if not root:
            return

        self.prepare(root.left, data)
        self.prepare(root.right, data)

        data.append(root.val)    